fix(tests): add a way to disable client metrics stderr

vladikoff opened the pull request:
Fixes #3158

vladikoff commented:
@shane-tomlinson r?

shane-tomlinson commented on server/lib/configuration.js:
```diff
@@ -48,6 +48,13 @@ var conf = module.exports = convict({
     doc: 'The location of the SSL certificate in pem format'
   },
+  client_metrics: {
+    enabled: {
```
`client_metrics.enabled` implies a global on/off switch of client metrics, whereas this flag only enables/disables output to stderr. Maybe this should be something like:

```js
client_metrics: {
  stderr_collector_enabled: {
     ...
  },
  statsd_collector_enabled: {
    ...
  }
}
```

I could also see something like: 
```js
client_metrics: {
  collectors: {
     default: [ 'ga', 'stderr', 'statsd' ],
     doc: 'Where to send client metrics'
  }
}
```

shane-tomlinson commented:
> shane-tomlinson was assigned by rfk 4 hours ago

I left my comments, back to @vladikoff

vladikoff commented:
@shane-tomlinson I updated this to be more consistent with the "DISABLE_..." options we already have. Also added some tests. If we need to mute statsd and other logging we can add that later if required, following the same pattern.

vladikoff commented:
uh metrics-unit keeps failing on node 0.10, will need to check what is going on, otherwise should be ready.

vladikoff commented:
Updated and should be ready!

shane-tomlinson commented:
https://travis-ci.org/mozilla/fxa-content-server/jobs/90769425 MUCH cleaner! Now if we could just get rid of all *this* cruft:

```
[2015-11-12 16:49:54] bid.ccverifier.INFO: compute cluster: spawned new worker process (3882) 1/20 processes running
[2015-11-12 16:49:54] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3882
[2015-11-12 16:49:54] bid.ccverifier.INFO: compute cluster: spawned new worker process (3884) 2/20 processes running
[2015-11-12 16:49:54] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3884
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 334
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3882 completed work in 0.35s
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 344
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3884 completed work in 0.33s
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3882
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 75
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3882 completed work in 0.08s
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: passing compute job to process 3882
[2015-11-12 16:49:55] bid.v2.INFO: assertion_verification_time 63
[2015-11-12 16:49:55] bid.server.INFO: 127.0.0.1 - - [Thu, 12 Nov 2015 16:49:55 GMT] "POST /v2 HTTP/1.1" 200 339 "-" "-"
[2015-11-12 16:49:55] bid.v2.INFO: verify { result: 'success', rp: 'http://127.0.0.1:9010' }
[2015-11-12 16:49:55] bid.ccverifier.DEBUG: compute cluster: process 3882 completed work in 0.07s
```
